spring-cloud-aws-autoconfigure/src/main/java/io/awspring/cloud/autoconfigure/s3/S3AutoConfiguration.java

```diff
@@ -116,6 +117,10 @@ S3Presigner s3Presigner(S3Properties properties, AwsProperties awsProperties,
 		else if (awsProperties.getEndpoint() != null) {
 			builder.endpointOverride(awsProperties.getEndpoint());
 		}
+		else if (connectionDetails.getIfAvailable() != null
```

Comment:
Shouldn't connection details be applied with highest priority like [here](https://github.com/awspring/spring-cloud-aws/blob/main/spring-cloud-aws-autoconfigure/src/main/java/io/awspring/cloud/autoconfigure/core/AwsClientBuilderConfigurer.java#L88-L92) ?
```suggestion
         if (connectionDetails.getIfAvailable() != null
             && connectionDetails.getIfAvailable().getEndpoint() != null) {
			builder.endpointOverride(connectionDetails.getIfAvailable().getEndpoint());
		}
		else if (properties.getEndpoint() != null) {
			builder.endpointOverride(properties.getEndpoint());
		}
		else if (awsProperties.getEndpoint() != null) {
			builder.endpointOverride(awsProperties.getEndpoint());
		}
```